app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee.erb

```diff
@@ -250,15 +254,30 @@ ShortFormApplicationService = ($translate, $http, Upload, uuid, ListingService,
     return (liveInSfEligible || workInSfEligible || neighborhoodResidenceEligible || displacedOrCertOfPreference)
 
   Service._updatePreference = (preference, eligibleMembers) ->
-    # only need to update if nobody is eligible
-    Service.preferences[preference] = false if _.isEmpty(eligibleMembers)
+    members = eligibleMembers.map (member) ->
+      Service._householdMemberName(member)
+    selectedMember = Service.preferences[preference + '_household_member']
+
+    # only need to clear if nobody is eligible or if selected householdMember is no longer eligible
+    if _.isEmpty(members) || !_.includes(members, selectedMember)
```

Comment:
we must have run into the same bug because I added a similar change to `_updatePreference` in the branch I'm working on -- we can combine when we get there.